### PR TITLE
chore: revert #5069 to restore Netlify builds

### PR DESCRIPTION
**Pull request by voxpelli**
`assetgraph-builder` is used through at least the `buildProduction` command

This reverts commit 53a4bafbdeb32576440b0a21787f2525585411c0.

## PR Checklist

- [x] ~~Addresses an existing open issue: fixes #000~~
- [x] ~~That issue was marked as [`status: accepting prs`](https://github.com/mochajs/mocha/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)~~
- [x] Steps in [CONTRIBUTING.md](https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md) were taken

## Overview

Reverts #5069 to restore Netlify builds by adding back `assetgraph-builder`

Downside: Installs on non-x86 machines again require `--ignore-scripts`: https://github.com/mochajs/mocha/issues/4828#issuecomment-1035204506

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,7 +81,9 @@
     "@rollup/plugin-json": "^4.1.0",
     "@rollup/plugin-multi-entry": "^4.0.1",
     "@rollup/plugin-node-resolve": "^13.1.3",
+    "assetgraph-builder": "^9.0.0",
     "autoprefixer": "^9.8.6",
+    "canvas": "^2.11.2",
```
Comment (JoshuaKGoldberg) [inline package.json:86]: Do we need to add `canvas` back too? Can we just revert the `assetgraph-builder` removal?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
I think we can keep the removal of `canvas`? Otherwise LGTM (alas)

File: package.json (hunk 1 above)
Comment (voxpelli) [reply package.json:86]: `assetgraph-builder` includes the `canvas` as well, so it didn't make any difference in terms of annoyance

Comment (JoshuaKGoldberg) [reply package.json:86]: Blurgh, 👍

**Review by JoshuaKGoldberg — APPROVED** [review]

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Or, actually, wait - `docs/*` doesn't need to be reverted. It doesn't need to use `canvas`.

**Pull request merged**